evaluation_wrapper.py

```diff
@@ -97,18 +109,85 @@ def plotOKS(self, epochs):
     def plotPCK(self, epochs):
         pass
 
+    def _full_list_of_predictions_wrapper(self, gen, model_sub_dir, epoch, average_flip_prediction=False):
+        print('Predicting over all batches...')
+        image_ids, list_of_predictions = self._full_list_of_predictions(gen, self.model_sub_dir, self.epoch, predict_using_flip=False)
+        print()
+
+        if average_flip_prediction:
+            print('Predicting over all batches using a horizontally flipped input, with prediction coordinates transformed back...')
+            image_ids_2, list_of_predictions_2 = self._full_list_of_predictions(gen, self.model_sub_dir, self.epoch, predict_using_flip=True)
+            print()
+
+            assert image_ids == image_ids_2, "Expected the image IDs should be in the same order"
+            eps = 1e-3
+            for i in range(len(list_of_predictions)):
+                # Average predictions from original image and the untransformed flipped image to get a more accurate prediction
+                for j in range(NUM_COCO_KEYPOINTS):
+                    base = j * NUM_COCO_KP_ATTRBS
+
+                    n = 0
+                    x_sum = 0
+                    y_sum = 0
+
+                    if list_of_predictions[i]['keypoints'][base+2] == 1:
+                        x_sum += list_of_predictions[i]['keypoints'][base]
+                        y_sum += list_of_predictions[i]['keypoints'][base + 1]
+                        n += 1
+
+                    if list_of_predictions_2[i]['keypoints'][base+2] == 1:
+                        x_sum += list_of_predictions_2[i]['keypoints'][base]
+                        y_sum += list_of_predictions_2[i]['keypoints'][base + 1]
+                        n += 1
+
+                    if n > 0:
+                        list_of_predictions[i]['keypoints'][base] = round(x_sum / n)
+                        list_of_predictions[i]['keypoints'][base + 1] = round(y_sum / n)
+                        list_of_predictions[i]['keypoints'][base + 2] = 1
```

Comment:
there's prob some numpy way to do this but I don't think it's worthwhile to figure it out